CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ cmake_minimum_required(VERSION 2.8.11)
 set(USER_CONFIG "${CMAKE_CURRENT_SOURCE_DIR}/config/user.cmake" CACHE PATH
   "Path to optional user configuration file.")
 
+set (CMAKE_CXX_STANDARD 11)
```

Comment:
I'm not sure when this was added, but I think the new "accepted" way to do this is using [`target_compile_features()`](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards).

Reply:
Totally fine with me if you want to change it. Beyond Appveyor, I don't think the CMake builds in this branch have been tested much.

Reply:
We should have @acfisher comment before I make the change. He's better with CMake and will be able to tell us what is more commonly used at LLNL.

Reply:
Sure I used CMAKE once, I'm totally an expert now ;).

Reply:
@jdahm, do you want to take another look at this? As I said, I really have no preference, except the simpler the better :)

Reply:
I pushed a slight extension to this. It's quite non-modern CMake, but it has support for almost all versions. If we want to require 3.1+ then we can start to require per-target "features" from this list:
https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html
Among them are variadic templates, etc. However, to keep things simple I think what was there with the minor extension I pushed is fine 👍

Reply:
Thank you Johann!